src/main/java/uk/gov/companieshouse/account/validator/service/handler/request/status/RequestStatusHandler.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.companieshouse.account.validator.service.handler.request.status;
+
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.account.validator.model.felix.ixbrl.Results;
+import uk.gov.companieshouse.account.validator.model.validation.RequestStatus;
+import uk.gov.companieshouse.account.validator.repository.RequestStatusRepository;
+import uk.gov.companieshouse.api.model.felixvalidator.ValidationStatusApi;
+
+@Component
+public class RequestStatusHandler {
+
+    private final RequestStatusRepository statusRepository;
+
+    @Autowired
+    public RequestStatusHandler(
+            RequestStatusRepository statusRepository) {
+        this.statusRepository = statusRepository;
+    }
+
+    public RequestStatus pending(String fileId, String fileName, ValidationStatusApi status) {
+        Results results = new Results();
+        results.setValidationStatus(status);
+        LocalDateTime createdDateTime = getCreatedDateTime(fileId);
+        // Either it does not exist - set to createdDate or set new localDateTime
+        LocalDateTime modifiedDateTime = isModifiedDateTimePresenter(fileId) ? LocalDateTime.now() : createdDateTime;
```

Comment:
I think two calls to the mongo here is bit unneccessary, the updated date is always LocalDateTime.now(). Something like: 
```suggestion
        LocalDateTime now = LocalDateTime.now();
        LocalDateTime createdDateTime = statusRepository.findById(fileId)
                .map(RequestStatus::createdDateTime)
                .orElse(now);
        LocalDateTime modifiedDateTime = now;
```